### PR TITLE
Use Briles/gruvbox .tmTheme files instead of peaceant's port

### LINKED ISSUE #1278: Use Briles/gruvbox .tmTheme files instead of peaceant's port
The last commit that included the .tmTheme files
in Briles/gruvbox was Briles/gruvbox@ec4eae233b

For convenience, I have cloned the .tmTheme files
into a separate repo [here](https://github.com/subnut/gruvbox-tmTheme)

I recommend using these .tmThemes because these
include all the variants (dark, light, hard, soft, medium, etc),
whereas peaceant's port does not

I would have done it myself, but I couldn't find any
guide to add themes to this repo....

### PR DESCRIPTION
**Pull request by j0hnmeow**
Closes #1278

### CODE REVIEW

File: tests/assets.rs
Comment (sharkdp) [inline tests/assets.rs:57]: Phew. That's an awful lot of different themes. To be honest, I would really prefer if they are not all part of `bat`. Light/dark variants are fine, but 23 different variations?!

**Review by sharkdp — CHANGES_REQUESTED** [review]
Thank you for your contribution!

File: tests/assets.rs
Comment (j0hnmeow) [reply tests/assets.rs:57]: Yes, it's a lot. Thought you could ask. There are 2 ways:
1. Removing some of them (e.g. `ndc` ones, there are not a part of official gruvbox)
2. Changing the test to check if there are some themes without checking for other ones (assets could have some unchecked themes)
Also 6 of them are symlinks

Comment (sharkdp) [reply tests/assets.rs:57]: > Changing the test to check if there are some themes without checking for other ones (assets could have some unchecked themes)

Well, I don't care too much about how the test code looks like, but if we add 23 gruvbox themes,  the `bat --list-themes` output will be completely overwhelmed.

Comment (j0hnmeow) [reply tests/assets.rs:57]: That's true. I believe `gruvbox`, `gruvbox-dark` and `gruvbox-light` is enough.

Comment (j0hnmeow) [reply tests/assets.rs:57]: Either this pr is invalid and bat should keep three variants or bat should learn how to work with theme variants

Comment (sharkdp) [reply tests/assets.rs:57]: @subnut?

Comment (subnut) [reply tests/assets.rs:57]: > Removing some of them (e.g. ndc ones, there are not a part of official gruvbox)

I am perfectly fine with it. Actually, I also had thought _what's the NDC for?_ but was unable to find anything, so I just included them in the repo anyway......

@j0hnmeow I think we should remove the NDC ones from the repo. Please push a commit at subnut/gruvbox-tmTheme#1 that removes all the NDC tmThemes

<br>

> Either this pr is invalid and bat should keep three variants or bat should learn how to work with theme variants

I think I will go with _bat should learn how to work with theme variants_ :smile: 

@sharkdp Just a suggestion for the future.... I can only see requests for new themes to be included in `bat` to keep growing as bat becomes more popular. So, I think, at some point, you should port the themes to some separate repo, say `bat-themes`, (yes, I am taking references from `vim-airline`) and include some mechanism to auto-build the theme cache when `bat` is updated. (Currently we need to manually re-build the cache ourselves) (This could be done by keeping all the tmThemes in a folder specified in the config file, so that when bat detects that the cache is invalid, it automatically re-builds the cache by using all the tmTheme files from that directory)

<br>

> if we add 23 gruvbox themes, the bat --list-themes output will be completely overwhelmed

I have a suggestion... Make `bat --list-themes` do just that, i.e. **list** the themes
Then the user could use a new `--preview-themes` to see the _preview_ of the themes 

Or, alternatively, implement `--preview-theme <theme_name>` to see the preview of <theme_name>, and show preview of all themes when `--preview-all-themes`

Comment (j0hnmeow) [reply tests/assets.rs:57]: > > if we add 23 gruvbox themes, the bat --list-themes output will be completely overwhelmed
> 
> I have a suggestion... Make `bat --list-themes` do just that, i.e. **list** the themes
> Then the user could use a new `--preview-themes` to see the _preview_ of the themes
> 
> Or, alternatively, implement `--preview-theme <theme_name>` to see the preview of <theme_name>, and show preview of all themes when `--preview-all-themes`

There's something similar in [dandavison/delta](https://github.com/dandavison/delta)

**Review by subnut — CHANGES_REQUESTED** [review]
As there is no difference between `gruvbox` and `gruvbox-dark` and `gruvbox-dark-medium`, all three should not stay.
I vote to keep `gruvbox-dark-medium` , because it explains the difference between this and.. say, `gruvbox-dark-hard`. 
Whereas, users are going to have a hard time trying to differentiate between `gruvbox` and `gruvbox-dark-hard`
<br>

The same analogy applies to `gruvbox-light`, `gruvbox-hard`, `gruvbox-medium`, `gruvbox-soft`, etc..

File: tests/assets.rs
Comment (subnut) [inline tests/assets.rs:45]: As there is no difference between `gruvbox` and `gruvbox-dark` and `gruvbox-dark-medium`, all three should not stay.
I vote to keep `gruvbox-dark-medium` , because it explains the difference between this and.. say, `gruvbox-dark-hard`. 
Whereas, users are going to have a hard time trying to differentiate between `gruvbox` and `gruvbox-dark-hard`
<br>

The same analogy applies to `gruvbox-light`, `gruvbox-hard`, `gruvbox-medium`, `gruvbox-soft`, etc..

**Review by subnut — CHANGES_REQUESTED** [review]

**Comment by subnut** [comment]
Now that the number of _new_ themes is reduced to three (total is six, but older three are being removed) and all of the six variants of the original gruvbox theme have been covered, I think this PR is ready to merge.

**Comment by subnut** [comment]
@sharkdp Anything keeping this PR from getting merged?

**Comment by sharkdp** [comment]
> @sharkdp Anything keeping this PR from getting merged?

1. the fact that I don't really understand why this is necessary and
2. that I don't really like the larger number of themes, even if it's just six.

That being said, I'm not completely opposed to this. I'll let the other two maintainers decide.

**Comment by subnut** [comment]
> the fact that I don't really understand why this is necessary

Consistency. See this -
image

**Comment by subnut** [comment]
> I don't really like the larger number of themes, even if it's just six.

@sharkdp Actually I thought that it would be better to include all the official variants of the theme, just for completeness. But if you don't like it, then I suggest you that you do this
| Current theme of this PR | Rename to |
|-|-|
| gruvbox-light-medium | gruvbox-light |
| gruvbox-dark-medium | gruvbox-dark |

And there you go! Only two themes, instead of the previous _three_!

**Comment by subnut** [comment]
@j0hnmeow Can you do the above yourself, or do I need to rename the files in my repo too?
If renaming is needed in my repo too, please send a PR to the [`bat-source`](https://github.com/subnut/gruvbox-tmTheme/tree/bat-source) branch.

**Comment by j0hnmeow** [comment]
> @j0hnmeow Can you do the above yourself, or do I need to rename the files in my repo too?
> If renaming is needed in my repo too, please send a PR to the [`bat-source`](https://github.com/subnut/gruvbox-tmTheme/tree/bat-source?rgh-link-date=2020-10-30T03%3A18%3A33Z) branch.

Yeah your repo needs renaming too

**Comment by subnut** [comment]
> Yeah your repo needs renaming too

@j0hnmeow 
Send a PR then, renaming -
|From|To|
|-|-|
| gruvbox-light-medium | gruvbox-light |
| gruvbox-dark-medium | gruvbox-dark |

And deleting the rest of the .tmThemes
Just make sure that you send the PR to the `bat-source` branch, okay?

**Comment by subnut** [comment]
@j0hnmeow What happened? Where's the PR?

**Comment by subnut** [comment]
@sharkdp Now there are **two** themes only, instead of the previous **three**

**Comment by sharkdp** [comment]
Yes - thank you for the update!

Two more things:
- Could you please update the CHANGELOG (see `CONTRIBUTING.md`)?
- Could you please remove `syntaxes.bin` from the PR. No syntaxes have been changed in this PR.

**Review by sharkdp — APPROVED** [review]

**Comment by subnut** [comment]
Ping @j0hnmeow

**Comment by j0hnmeow** [comment]
> Yes - thank you for the update!
> 
> Two more things:
> 
>     * Could you please update the CHANGELOG (see `CONTRIBUTING.md`)?
> 
>     * Could you please remove `syntaxes.bin` from the PR. No syntaxes have been changed in this PR.

Done

**Comment by subnut** [comment]
Ping @sharkdp

**Comment by sharkdp** [comment]
This PR now shows a lot of unrelated changes.

**Comment by subnut** [comment]
@j0hnmeow Will you work on it or should I take it up (in another PR) ?

**Comment by j0hnmeow** [comment]
I will

**Comment by j0hnmeow** [comment]
@sharkdp done (took a while)

**Comment by subnut** [comment]
Ping @sharkdp

**Comment by j0hnmeow** [comment]
It's been over 2 months btw

**Comment by sharkdp** [comment]
The changelog entry has been added to `v0.17.0`, while it should be added to the "unreleased" section on the top. We should also mention that this is a *breaking change* for everyone using "gruvbox" so far (they need to switch to "gruvbox-dark" or "gruvbox-light").

**Comment by j0hnmeow** [comment]
> We should also mention that this is a _breaking change_ 
How should that be done?

**Comment by sharkdp** [comment]
I added a comment to the CHANGELOG. That will be part of the next release notes. Thank you!

**Comment by j0hnmeow** [comment]
Wow, after almost 3mo it's finally done?

**Comment by sharkdp** [comment]
> It's been over 2 months btw

> Wow, after almost 3mo it's finally done?

If that's merely an observation, alright. If there is a sarcastic undertone, I don't appreciate it. This is a project that is maintained by volunteers, not a corporate product with deadlines.

Not that I need to explain myself, but in case you are curious:

This change here **is not**:
* a critical bugfix that I would like to get out as quick as possible
* a cool new feature that I would like to use myself
* something I am personally particularly interested in

On the other hand, **it is**:
* a breaking change.. which means that I am extra cautious
* a lot of work. I had to write multiple comments until it was finally in a form to merge.

Look. I am super grateful for every contribution. I am also very grateful for this contribution and for the work that both of you did to get this included. But I don't appreciated being rushed or urged. And I definitely don't like being ridiculed.

**Comment by j0hnmeow** [comment]
It's just an observation with some self-irony. I'm sorry if you find it ridiculing or offending, I didn't mean that. I respect your hard work, it's unlikely for me to make at least as big project myself.

**Comment by sharkdp** [comment]
Added in [bat v0.18](https://github.com/sharkdp/bat/releases/tag/v0.18.0).

**Pull request merged**
